expandFns.php

```diff
@@ -320,6 +320,25 @@ function tidy_date($string) {
   $string=trim($string);
   if (stripos($string, 'Invalid') !== FALSE) return '';
   if (!preg_match('~\d{2}~', $string)) return ''; // If there are not two numbers next to each other, reject
+  // Huge amout of character cleaning
+  if (strlen($string) != mb_strlen($string)) {  // Convert all multi-byte characters to dashes
+    $cleaned = '';
+    for ($i = 0; $i < mb_strlen($string); $i++) {
+       $char = mb_substr($string,$i,1);
+       if (mb_strlen($char) == strlen($char)) {
+          $cleaned .= $char;
+       } else {
+          $cleaned .= '-';
+       }
+    }
+    $string = $cleaned;
+  }
+  $string = preg_replace("~[^\x01-\x7F]~","-", $string); // Convert any non-ASCII Characters to dashes
+  $string = preg_replace('~[\s\-]*\-[\s\-]*~', '-',$string); // Combine dash with any following or preceeding white space and other dash
+  $string = preg_replace('~\-$~', '',$string);  // Remove trailing dash
```

Comment:
Could we catch multiple dashes (and combine two pregs into one) with:
```suggestion
  $string = preg_replace('~^\-*(.+?)\-*$~', '\1', $string);  // Remove trailing dash
```